### PR TITLE
Add routes

### PR DESCRIPTION
**Pull request by gene83**

### CODE REVIEW

**Review by melsoriano — APPROVED** [review]
LGTM ✅

File: server/routes/api/contacts/index.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+  Contact.where('created_by', userId)
+    .fetchAll()
+    .then(contacts => {
+      if (contacts === null) {
+        return res.json({});
+      }
+
+      contacts = contacts.toJSON();
+
+      const filteredContacts = contacts.filter(contact => {
```
Comment (taesup) [inline server/routes/api/contacts/index.js]: You should try to use the DB to filter the contacts, it'll be faster.

File: server/routes/api/contacts/index.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    });
+});
+
+router.delete('/:id', isAuthenticated, (req, res) => {
+  const contactId = req.params.id;
+
+  Contact.where('id', contactId)
+    .destroy()
+    .then(() => {
+      res.status(200);
```
Comment (taesup) [inline server/routes/api/contacts/index.js]: You don't need to do a status 200. `res.json` will set it to 200 for you.

File: server/routes/api/index.js [hunk 3]
```diff
@@ -1,4 +1,123 @@
...
+  const userId = req.user.id;
+  const { username, name, email, address } = req.body;
+  const editedUserReq = { username, name, email, address };
+
+  User.where('id', userId)
+    .fetch()
+    .then(dbUser => {
+      dbUser
+        .save(editedUserReq)
+        .then(resUser => {
```
Comment (taesup) [inline server/routes/api/index.js]: You should return the dbUser, then the `.then` and the `.catch` here can be part of the upper promise chain.

**Review by taesup — CHANGES_REQUESTED** [review]

**Review by taesup — APPROVED** [review]

**Pull request merged**